nginx_config_reloader/__init__.py

```diff
@@ -78,8 +79,19 @@ def process_IN_MOVE_SELF(self, event):
         raise ListenTargetTerminated
 
     def handle_event(self, event):
+        queue_length = len(self.notifier._eventq)
+        if queue_length > 1:
```

Comment:
You should check that the other events in the queue are not going to be ignored, right?

Reply:
what other events? on the first event there will be one event in the queue and it will then handle the event. then once events are consumed it eventually goes down (I think the resolution isn't there to guarantee that it will again be exactly 1 consuming) but at least as long as the first event is consumed (and our reasonable 5 second sleep is long enough) the change will be reloaded

Reply:
this is a deque() filled with unique events (due to `coales`) settings: https://github.com/seb-m/pyinotify/blob/0f3f8950d12e4a6534320153eed1a90a778da4ae/python3/pyinotify.py#L1245

seems to have the answer, but it's a "private" attr (prefixed with `_`) so change could be expected.
I suggest we use fallback mode (if attr is renamed, or changed type, etc) to make sure code won't break out of bounds.

Reply:
another approach could be to avoid looking into the notifier's queue, and try to find the last event, by checking the [check_events](https://github.com/seb-m/pyinotify/blob/0f3f8950d12e4a6534320153eed1a90a778da4ae/python3/pyinotify.py#L1176) method of the notifier.
then we don't use private `_eventq` anymore, only public API, and we act on the last event, which could avoid reloading Nginx while configs are being changed.
I'm not sure if it's doable though.

Reply:
yup, cool. works. see https://github.com/ByteInternet/nginx_config_reloader/pull/41/commits/89310a5c76469699589fc40e33164361eec4e618